packages/blitz-next/src/index-server.ts

```diff
@@ -187,8 +187,8 @@ export const setupBlitzServer = ({plugins, onError}: SetupBlitzOptions) => {
     ): NextApiHandler<TResult | void> =>
     async (req, res) => {
       try {
-        await handleRequestWithMiddleware(req, res, middlewares)
-        return handler(req, res, res.blitzCtx)
+        middlewares.push((req, res) => handler(req, res, res.blitzCtx));
+        return await handleRequestWithMiddleware(req, res, middlewares);
```

Comment:
In order to have it typed properly we can do this:
```suggestion
     return await handleRequestWithMiddleware(req, res, [...middlewares, (req, res) => handler(req, res, res.blitzCtx)]);
```
The `handleRequestWithMiddleware` will infer the types properly based on the third argument.

Reply:
oh yeah, great idea! Thank you!